packages/effect/dtslint/Struct.ts

```diff
@@ -1,8 +1,157 @@
 import { pipe } from "effect/Function"
 import * as S from "effect/Struct"
 
+const asym = Symbol.for("effect/dtslint/a")
+const bsym = Symbol.for("effect/dtslint/b")
+const csym = Symbol.for("effect/dtslint/c")
+const dsym = Symbol.for("effect/dtslint/d")
+
+declare const stringNumberRecord: Record<string, number>
+declare const symbolNumberRecord: Record<symbol, number>
+declare const numberNumberRecord: Record<number, number>
+
+const stringStruct = { a: "a", b: 1, c: true }
+const symbolStruct = { [asym]: "a", [bsym]: 1, [csym]: true }
+const numberStruct = { 1: "a", 2: 1, 3: true }
+
```

Comment:
I think two more interesting test cases are:

* Using template strings in index signatures:
    ```ts
    declare const templateStringNumberRecord: Record<`a${string}`, number>
    
    // $ExpectType number | undefined
    pipe(templateStringNumberRecord, S.get("a"))
    ```
* Using an intersection of a struct and record:
     ```ts
    declare const structRecordIntersection: Record<string, number> & { a: number };
    
    // $ExpectType number
    pipe(structRecordIntersection, S.get("a"))

    // @ts-expect-error
    pipe(structRecordIntersection, S.get("b"))
    ```